### PR TITLE
ci: cleanup `cxx14` and `cxx20` Dockerfiles

### PR DESCRIPTION
**Pull request by coryan**
Reduce inconsistencies between these Dockerfiles and between the different steps in the Dockerfiles themselves.

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/12725)

### CODE REVIEW

File: ci/cloudbuild/dockerfiles/fedora-latest-cxx14.Dockerfile [hunk 1]
```diff
@@ -57,32 +57,30 @@ RUN curl -fsSL https://pkgconfig.freedesktop.org/releases/pkg-config-0.29.2.tar.
     ldconfig
 ENV PKG_CONFIG_PATH=/usr/local/lib64/pkgconfig:/usr/local/lib/pkgconfig:/usr/lib64/pkgconfig
 
-# Install Abseil, remove the downloaded files and the temporary artifacts
-# after a successful build to keep the image smaller (and with fewer layers).
+# Download and install direct dependencies of `google-cloud-cpp`. Including
+# development dependencies.  In each case, remove the downloaded files and the
+# temporary artifacts after a successful build to keep the image smaller (and
+# with fewer layers)
```
Comment (dbolduc) [inline ci/cloudbuild/dockerfiles/fedora-latest-cxx14.Dockerfile:63]: nit: `s/$/./`

File: ci/cloudbuild/dockerfiles/fedora-latest-cxx20.Dockerfile [hunk 2]
```diff
@@ -59,23 +59,24 @@ RUN curl -fsSL https://pkgconfig.freedesktop.org/releases/pkg-config-0.29.2.tar.
     ldconfig
 ENV PKG_CONFIG_PATH=/usr/local/lib64/pkgconfig:/usr/local/lib/pkgconfig:/usr/lib64/pkgconfig
 
-# Install Abseil, remove the downloaded files and the temporary artifacts
-# after a successful build to keep the image smaller (and with fewer layers)
+# Download and install direct dependencies of `google-cloud-cpp`. Including
+# development dependencies.  In each case, remove the downloaded files and the
+# temporary artifacts after a successful build to keep the image smaller (and
+# with fewer layers)
```
Comment (dbolduc) [inline ci/cloudbuild/dockerfiles/fedora-latest-cxx20.Dockerfile:65]: and here

**Review by dbolduc — APPROVED** [review]
if you don't feel like pushing another commit, that would be fine with me.

File: ci/cloudbuild/dockerfiles/fedora-latest-cxx14.Dockerfile (hunk 1 above)
Comment (coryan) [reply ci/cloudbuild/dockerfiles/fedora-latest-cxx14.Dockerfile]: Done.

File: ci/cloudbuild/dockerfiles/fedora-latest-cxx20.Dockerfile (hunk 2 above)
Comment (coryan) [reply ci/cloudbuild/dockerfiles/fedora-latest-cxx20.Dockerfile]: Done.

**Pull request merged**